src/KafkaFactory.ts

```diff
@@ -0,0 +1,33 @@
+import kafkaDojot = require("@dojot/adminkafka");
+import { logger } from "@dojot/dojot-module-logger";
+import kafka = require("kafka-node");
+import config = require("./config");
+
+export class KafkaFactory {
+    private host?: string;
+
+    constructor(theHost?: string) {
+        this.host = theHost;
+    }
+
+    public client(): kafka.KafkaClient {
+        logger.debug("Creating new Kafka producer...", {filename: "producer"});
+        const kafkaHost = this.host ? this.host : config.kafka.kafkaAddress + ":" + config.kafka.kafkaPort.toString();
+        logger.debug("Creating Kafka client...", {filename: "producer"});
+        return new kafka.KafkaClient({kafkaHost});
+    }
+
+    public dojotProducer(): kafkaDojot.Admin {
+        logger.debug("... Kafka client was created.", {filename: "producer"});
+        logger.debug("Creating Kafka HighLevenProducer...", {filename: "producer"});
+        return new kafkaDojot.Admin(config.kafka.kafkaAddress, Number(config.kafka.kafkaPort));
```

Comment:
Also, it might be a good thing to check whether node-rdkafka already offers all features that adminkafka has (which is solely used to create topics). If so, this could be a good time to make the switch :)

Reply:
I agree, but I won't change right now, it could be the next task in the near future, once we want to release a version and we want to minimize the changes.

Reply:
Hi @giovannicuriel  and @raulnegreiros. I think this might be a good idea @giovannicuriel, but I agree with you @raulnegreiros. We need to release a version right now and we want to minimize the changes.